text-minimessage/src/main/java/net/kyori/adventure/text/minimessage/translation/MiniMessageTranslator.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2022 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.text.minimessage.translation;
+
+import java.text.MessageFormat;
+import java.util.Locale;
+import java.util.Objects;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.TranslatableComponent;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import net.kyori.adventure.translation.Translator;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public abstract class MiniMessageTranslator implements Translator {
+
+  final MiniMessage miniMessage;
+
+  public MiniMessageTranslator() {
+    this(MiniMessage.miniMessage());
+  }
+
+  public MiniMessageTranslator(final @NotNull MiniMessage miniMessage) {
+    this.miniMessage = Objects.requireNonNull(miniMessage, "miniMessage");
+  }
+
+  protected abstract @Nullable String getMiniMessageString(final @NotNull String key, final @NotNull Locale locale);
+
+  @Override
+  public final @Nullable MessageFormat translate(final @NotNull String key, final @NotNull Locale locale) {
+    return null;
+  }
+
+  @Override
+  public @Nullable Component translate(final @NotNull TranslatableComponent component, final @NotNull Locale locale) {
+    final String miniMessageString = getMiniMessageString(component.key(), locale);
+
+    if (miniMessageString == null) {
+      return null;
+    }
+
+    final Component resultingComponent;
+
+    if (component.args().isEmpty()) {
+      resultingComponent = MiniMessage.miniMessage().deserialize(miniMessageString);
+    } else {
+      resultingComponent = MiniMessage.miniMessage().deserialize(miniMessageString, new ArgumentTag(component.args()));
+    }
```

Comment:
Supports this tags in tags. Like property references ?

Ex: `<lang:FANCY_LANG_KEY:'<lang:FANCY_LANG_PREFIX>'>`

Reply:
Yes it should do